### ISSUE TITLE
What is the idiomatic way to get training loss from `gradient()`?

### ISSUE THREAD

**Issue by karlfroldan** [FluxML/Flux.jl#1950]
In my current knowledge, appending the training loss to a list inside `gradient() do end` does not work since `push!` is non-differentiable. My current workaround is recomputing the whole training loss after computing the gradients which seems redundant to me and it wastes time and resources.

Is there a current and idiomatic way to get the training loss after computing `gradient()`? If none, are there any plans on making this possible?

**Comment by ToucheSir** [comment]
https://fluxml.ai/Zygote.jl/latest/utils/#Zygote.withgradient, which you can access from the Flux namespace.

**Comment by karlfroldan** [comment]
> https://fluxml.ai/Zygote.jl/latest/utils/#Zygote.withgradient, which you can access from the Flux namespace.

Thanks! Didn't see this when scouring Zygote's source earlier.
